compositor_pipeline/src/pipeline/whip_whep/whip_handlers/create_whip_session.rs

```diff
@@ -0,0 +1,217 @@
+    offer: String,
+) -> Result<Response<Body>, WhipServerError> {
+    let input_id = InputId(Arc::from(id.clone()));
+
+    validate_sdp_content_type(&headers)?;
+
+    let input_components = get_whip_input_connection_options(state.clone(), input_id.clone())?;
+
+    validate_token(input_components.bearer_token, headers.get("Authorization")).await?;
+
+    //Deleting previous peer_connection on this input which was not in Connected state
+    if let Some(connection) = input_components.peer_connection {
+        if let Err(err) = connection.close().await {
+            return Err(WhipServerError::InternalError(format!(
+                "Cannot close previously existing peer connection {input_id:?}: {err:?}"
+            )));
+        }
+    }
+
+    let peer_connection = init_peer_connection(
+        input_components.video_sender.is_some(),
+        input_components.audio_sender.is_some(),
+    )
+    .await?;
+
+    if let Ok(mut connections) = state.input_connections.lock() {
+        if let Some(connection) = connections.get_mut(&input_id) {
+            connection.peer_connection = Some(peer_connection.clone());
+        } else {
+            return Err(WhipServerError::InternalError(
+                "Peer connection initialization error".to_string(),
+            ));
+        }
+    } else {
+        return Err(WhipServerError::InternalError(
+            "Cannot lock input connections".to_string(),
+        ));
+    }
+
+    peer_connection.on_track(Box::new(move |track, _, _| {
+        let track_kind = track.kind();
+        let state_clone = state.clone();
+        let input_id_clone = input_id.clone();
+        let video_sender_clone = input_components.video_sender.clone();
+        let audio_sender_clone = input_components.audio_sender.clone();
+        let depayloader_clone = input_components.depayloader.clone();
+
+        tokio::spawn(async move {
+            match track_kind {
+                RTPCodecType::Video => {
+                    if let Some(sender) = video_sender_clone {
+                        handle_track(
+                            track,
+                            state_clone,
+                            input_id_clone,
+                            depayloader_clone,
+                            sender,
+                        )
+                        .await;
+                    }
+                }
+                RTPCodecType::Audio => {
+                    if let Some(sender) = audio_sender_clone {
+                        handle_track(
+                            track,
+                            state_clone,
+                            input_id_clone,
+                            depayloader_clone,
+                            sender,
+                        )
+                        .await;
+                    }
+                }
+                _ => {
+                    debug!("RTPCodecType not supported!")
+                }
+            }
+        });
+        Box::pin(async {})
```

Comment:
why are you spawning new tokio task above, instead of doing all the work in `Box::pin(async {})`?